File: text-zoom/ios/Plugin/TextZoomPlugin.m
#import <Foundation/Foundation.h>
#import <Jigra/Jigra.h>

// Define the plugin using the JIG_PLUGIN Macro, and
// each method the plugin supports using the JIG_PLUGIN_METHOD macro.
JIG_PLUGIN(TextZoomPlugin, "TextZoom",
           JIG_PLUGIN_METHOD(getPreferred, JIGPluginReturnPromise);
)

File: text-zoom/ios/Plugin/TextZoomPlugin.swift
import Foundation
import Jigra

@objc(TextZoomPlugin)
public class TextZoomPlugin: JIGPlugin {
    private let textZoom = TextZoom()

    @objc func getPreferred(_ call: JIGPluginCall) {
        call.resolve([
            "value": textZoom.preferredFontSize()
        ])
    }
}
